### ISSUE TITLE
Support a `#[bench]`-like registration attribute macro

### ISSUE THREAD

**Issue by epage** [bheisler/criterion.rs#262]
The one thing holding me back from migrating is concern that I'll forget to update my list of tests when I write a whole bunch of new ones after being spoiled by `#[test]`.

With attribute macros and [`inventory`](https://crates.io/crates/inventory), we should be able to imitate `#[bench]`.

**Comment by bheisler** [comment]
Hey, thanks for the suggestion!

This is on the roadmap, yes, but not until after 0.3.0. It will use the new (currently nightly-only) custom test framework system as well.

**Comment by bheisler** [comment]
Duplicate of #205.
